lib/feature/home_page/widgets/pokemon_tile.dart

```diff
@@ -49,57 +52,62 @@ class _PokemonTileState extends State<PokemonTile> {
-              style: const TextStyle(
-                color: Colors.black,
-                fontWeight: FontWeight.bold,
+      child: Container(
+        decoration: BoxDecoration(
+          borderRadius: BorderRadius.circular(15),
+          color: thisTileTypes?.subTypes?.first.type?.name!.getPokemonColor ?? Colors.white,
+        ),
+        child: Stack(
+          children: [
+            Positioned(
+              left: 5,
+              top: 5,
+              child: Text(
+                widget.thisPokemon.name?.capitalize ?? '',
+                style: const TextStyle(
+                  color: Colors.white,
+                  fontWeight: FontWeight.bold,
+                  fontSize: 20,
+                ),
               ),
-              textAlign: TextAlign.left,
             ),
-            const SizedBox(height: 15),
-            Row(
-              children: [
-                Padding(
-                  padding: const EdgeInsets.fromLTRB(20, 1, 1, 1),
-                  child: Column(children: [
-                    Text('${thisTileTypes?.subTypes?.first.type?.name ?? ''}'),
-                    Text(
-                        '${thisTileTypes?.subTypes?.last.type?.name == thisTileTypes?.subTypes?.first.type?.name ? '' : thisTileTypes?.subTypes?.last.type?.name}'),
-                  ]),
-                ),
-                Container(
-                  width: 100,
-                  height: 100,
-                  alignment: Alignment.centerRight,
-                  child: Image.network(
-                    '$pokemonImgUrl${thisTileTypes?.id}.png',
-                    fit: BoxFit.cover,
-                    errorBuilder: (_, __, ___) {
-                      return CircularProgressIndicator();
-                    },
+            Positioned(
+              left: 5,
+              top: 35,
+              child: Column(
+                crossAxisAlignment: CrossAxisAlignment.start,
+                children: [
+                  PillContainerWidget(
+                    type: thisTileTypes?.subTypes?.first.type?.name,
+                    color: typeDetailsPageBackgroundColor,
                   ),
-                )
-              ],
+                  SizedBox(height: 5),
+                  Visibility(
+                    visible: thisTileTypes?.subTypes?.last.type?.name != thisTileTypes?.subTypes?.first.type?.name,
+                    child: PillContainerWidget(
+                      type: thisTileTypes?.subTypes?.last.type?.name ?? '',
+                      color: typeDetailsPageBackgroundColor,
+                    ),
+                  )
+                ],
+              ),
             ),
-          ]),
+            Positioned(
+              bottom: 0,
+              right: 0,
+              child: Image.network(
+                '$pokemonImgUrl${thisTileTypes?.id}.png',
+                width: 115,
+                height: 115,
+                errorBuilder: (_, __, ___) {
```

Comment:
use `loadingBuilder` for your loading widget and make a separate widget that would look good that represents the error, use that in your `errorBuilder`